### ISSUE TITLE
Invalid `NuGet.exe` commands should exit with a non-zero code

### ISSUE THREAD

**Issue by dougbu** [NuGet/Home#7981]
# Information
## Details about Problem
- NuGet product used: NuGet.exe
- NuGet version (x.x.x.xxx): 4.1.0.2450 / 4.1.0 and 5.0.0.5923 / 5.0.0+42a8779499c1d1ed2488c2e6b9e2ee6ff6107766 though likely all versions
- OS version (i.e. win10 v1607 (14393.321)): Windows 10 v1809 (OS build 17763.379) though problem does not appear specific to the OS version
- Worked before? **No**

## Detailed repro steps so we can see the same problem
1. Execute an incomplete or invalid NuGet.exe command
2. Observe error messages about the command line
3. Check the exit code

### Expected
Non-zero exit code

### Actual
Exit code `0`, meaning an Azure DevOps pipeline using this command incorrectly appears to _succeed_ ❕

### Verbose Logs
From a PowerShell session, reproducing issues we experienced in an Azure DevOps release pipeline.
```
> ...\nuget.exe push -timeout 1000
push: invalid arguments.
usage: NuGet push <package path> [API key] [options]

Pushes a package to the server and publishes it.
NuGet's default configuration is obtained by loading %AppData%\NuGet\NuGet.config, then loading any nuget.config or .nug
et\nuget.config starting from root of drive and ending in current directory.

     Specify the path to the package and your API key to push the package to the server.

options:

 -Source                  (src)  Specifies the server URL. If not specified, nuget.org is used unless DefaultPushSource
                                 config value is set in the NuGet config file.
 -ApiKey                         The API key for the server.
 -SymbolSource                   Specifies the symbol server URL. If not specified, nuget.smbsrc.net is used when pushi
                                 ng to nuget.org.
 -SymbolApiKey                   The API key for the symbol server.
 -Timeout                        Specifies the timeout for pushing to a server in seconds. Defaults to 300 seconds (5 m
                                 inutes).
 -DisableBuffering               Disable buffering when pushing to an HTTP(S) server to decrease memory usage. Note tha
                                 t when this option is enabled, integrated windows authentication might not work.
 -NoSymbols                      If a symbols package exists, it will not be pushed to a symbol server.
 -Help                    (?)    help
 -Verbosity                      Display this amount of details in the output: normal, quiet, detailed.
 -NonInteractive                 Do not prompt for user input or confirmations.
 -ConfigFile                     The NuGet configuration file. If not specified, file %AppData%\NuGet\NuGet.config is u
                                 sed as configuration file.
 -ForceEnglishOutput             Forces the application to run using an invariant, English-based culture.

examples:

nuget push foo.nupkg 4003d786-cc37-4004-bfdf-c4f3e8ef9b3a

nuget push foo.nupkg 4003d786-cc37-4004-bfdf-c4f3e8ef9b3a -src http://customsource/

nuget push foo.nupkg

nuget push foo.nupkg.symbols

nuget push foo.nupkg -Timeout 360

For more information, visit http://docs.nuget.org/docs/reference/command-line-reference

> $LASTEXITCODE
0
```

## See also
- https://github.com/NuGet/Home/issues/7224 may be related
- https://github.com/aspnet/AspNetCore-Internal/issues/2191 which is about AzDO issue we experienced attempting to add a timeout to the executed NuGet.exe commands

**Comment by nkolev92** [comment]
Did @dominoFire's PR https://github.com/NuGet/NuGet.Client/pull/2505 address this?

**Comment by nkolev92** [comment]
Duplicate of https://github.com/NuGet/Home/issues/7178. 

Fixed in 5.1
